### ISSUE TITLE
Annoying 'Note' displayed by function 'kronecker' about sparse matrices

### ISSUE THREAD

**Issue by pneuvial** [pneuvial/c3co#18]
```
> example(c3co)
Note: method with signature ‘dsparseMatrix#dsparseMatrix’ chosen for function ‘kronecker’,
target signature ‘dgCMatrix#dgTMatrix’.
"sparseMatrix#TsparseMatrix" would also be valid
```

**Comment by pneuvial** [comment]
Reproducible example:

```
W <- matrix(1:6, 3, 2)
L <- 5
p <- ncol(W)
Dm1 <- -Matrix::bandSparse(L, L, k=0:(L-1))
X <- kronecker(Dm1, W)
```

**Comment by pneuvial** [comment]
Can be fixed (e.g.) by converting W to a sparse matrix:

```
W <- matrix(1:6, 3, 2)
W <- as(W, "sparseMatrix")
L <- 5
p <- ncol(W)
Dm1 <- -Matrix::bandSparse(L, L, k=0:(L-1), giveCsparse = FALSE)
X <- kronecker(Dm1, W)
```
Fixed likewise in [develop cb8a136]
